keps/sig-cli/0031-kustomize-integration.md

```diff
@@ -167,30 +256,151 @@ To get the detailed objects in a kustomization
 kubectl get -f <dir> --dry-run -o yaml|json
 ```
 
-#### delete
+##### delete
 The command visible to users is
+
 ```
 kubectl delete -f <dir>
 ```
 
-### Implementation Details/Notes/Constraints
+#### Why we like this approach
+
+- It is capable of friction that requires deeper integration - such as producing errors referencing line
+  numbers of the original files (rather than the output files).
+- It integrates will all commands - get, delete, etc - would automatically work.
+- It is more consistent with UX workflow for commands that work directly off a target
+- It has a cleaner and simpler UX than Option A, fewer steps
+- It integrates the solution directly into the commands with issues instead of tacking it on as separate step
+- Users less likely to get confused and accidentally apply a directory with a kustomization.yaml and have
+  it do the wrong thing (it will do the right thing).
+
+#### Why don't like this approach
+
+- Kustomize does more than the resource builder currently does and this could surprise users
+- Commands are doing more than they were before
+
+### Why not as a kubectl plugin instead of compiled in?
+
+- The kubectl plugin mechanism does not provide a solution for distribution.  Because the functionality is intended as
+ the project's solution to issues within kubectl, we want it to be available to users of kubectl without additional
+ steps.  Having users manually download only Kustomize as a plugin might be ok, but it won't scale as a good approach
+ as the set of commands grows.
+- The effort to build and test the tool for all targets, develop a release process, etc. would be much higher for SIG
+  CLI, also, and it would exacerbate kubectl's version-skew challenges.
+- It will not support integration at more than a surface level - such as into the resource builder
+  (which does not offer a plugin mechanism).
+    - It was previously decided we didn't want to add a plugin mechanism to the resource builder.
+      This could be reconsidered, but would need to think through it more and figure out how to address
+      previously brought up issues.  There may be other issues not listed here as well.
+      - https://github.com/kubernetes/kubernetes/issues/13241
+      - https://github.com/kubernetes/kubernetes/pull/14993
+      - https://github.com/kubernetes/kubernetes/pull/14918
+- There is a risk that publishing each command as a separately built binary could cause the aggregate download
+  size of the toolset to balloon.  The kubectl binary is *52M* and the kustomize binary is *31M*.  (extrapolate to
+  30+ commands x 30MB).  Before going down this route, we should consider how to we might want to design a solution
+  and the tradeoffs.    
+
+**AI's for kubectl extensibility:**
```

Comment:
I'm not convinced these are meant to be here. They make sense in the overall kubectl sense, but not tide with kubectl extensions, maybe we should have a separate KEP that will be discussing just that and will outline the decision we'll make within our SIG.

Reply:
+1